types/document.d.ts

```diff
@@ -191,7 +193,7 @@ declare module 'mongoose' {
      * for immutable properties. Behaves similarly to `set()`, except for it
      * unsets all properties that aren't in `obj`.
      */
-    overwrite(obj: AnyObject): this;
+    overwrite(obj: Partial<Omit<DocType, '_id'>>): this;
```

Comment:
Similar here. `AnyObject` is more correct, `Partial<>` is wrong, and omitting `_id` is wrong.